myapp/tests/test_views.py

```diff
@@ -0,0 +1,108 @@
+import pytest
+from rest_framework.test import APIClient
+from myapp.models import Book
+from rest_framework_simplejwt.tokens import RefreshToken
+from django.contrib.auth import get_user_model
+from .factories import BookFactory, AuthorFactory, GenreFactory, PublisherFactory
+
+import os
+from django.core.files.uploadedfile import SimpleUploadedFile
+from django.conf import settings
+
+pytestmark = pytest.mark.django_db
+
+
+@pytest.fixture
+def api_client():
+    return APIClient()
+
+
+@pytest.fixture
+def create_books():
+    author = AuthorFactory()
+    genre = GenreFactory()
+    books = BookFactory.create_batch(5, authors=[author], genres=[genre])
+    return books
+
+
+
+@pytest.fixture
+def user():
+    """Fixture to create a user for authentication."""
+    User = get_user_model()
+    return User.objects.create_user(username="testuser", password="password123")
```

Comment:
Create and use the user factory

Reply:
Created user factories and used it for user auth